src/FastRoute.php

```diff
@@ -212,37 +214,29 @@ public function match(Request $request): MatchingResult
      *     in route used to generate URI.
      *
      * @return string URI path generated.
-     * @throws \RuntimeException if the route name is not known
-     *     or a parameter value does not match its regex.
+     * @throws \RuntimeException if the route name is not known or a parameter value does not match its regex.
      */
     public function generate(string $name, array $parameters = []): string
     {
         // Inject any pending routes
         $this->injectRoutes();
 
-        if (!array_key_exists($name, $this->routes)) {
-            throw new \RuntimeException(sprintf(
-                'Cannot generate URI for route "%s"; route not found',
-                $name
-            ));
-        }
-
-        $route = $this->routes[$name];
+        $route = $this->getRoute($name);
         $parameters = array_merge($route->getDefaults(), $parameters);
 
-        $defaultValues = [];
-        if (!empty($parameters['defaults'])) {
-            $defaultValues = $parameters['defaults'];
-        }
-
-        $routeParser = new RouteParser();
-        $routes = array_reverse($routeParser->parse($route->getPattern()));
+        /**
+         * TODO may loop be removed?
+         * what's it for?
+         * more than one route by name cannot exist
```

Comment:
Do you mean `foreach ($routes as $parts) {`?

Reply:
Yes, sure. $parsedRoutes always contains one element.

Reply:
Yes. Likely could be remove then.

Reply:
Removed.